pkg/statdb/statdb.go

```diff
@@ -184,14 +184,14 @@ func (s *Server) Update(ctx context.Context, updateReq *pb.UpdateRequest) (resp
 	}, nil
 }
 
-// UpdateBatch for updating  multiple farmers' stats in the db
+// UpdateBatch for updating multiple farmers' stats in the db
 func (s *Server) UpdateBatch(ctx context.Context, updateBatchReq *pb.UpdateBatchRequest) (resp *pb.UpdateBatchResponse, err error) {
-	// todo(moby) how should we handle one node failing to update but not all?
 	defer mon.Task()(&ctx)(&err)
 	s.logger.Debug("entering statdb UpdateBatch")
 
 	APIKeyBytes := updateBatchReq.APIKey
 	nodeStatsList := make([]*pb.NodeStats, len(updateBatchReq.NodeList))
```

Comment:
Define `nodeStatsList` the same way `failedNodes` is defined below. It will not have a length of `len(updateBatchReq.NodeList)` if any of the nodes fail